extensions/typescript-language-features/src/test/unit/previewer.test.ts

```diff
@@ -87,45 +87,92 @@ suite('typescript.previewer', () => {
 			'*@param* `parámetroConDiacríticos` — this will not');
 	});
 
+	test('Should render @example blocks as code', () => {
+		assert.strictEqual(
+			tagsMarkdownPreview([
+				{
+					name: 'example',
+					text: 'code();'
+				}
+			], noopToResource),
+			'*@example*  \n```\ncode();\n```'
+		);
+	});
+
+	test('Should not render @example blocks as code as if they contain a codeblock', () => {
+		assert.strictEqual(
+			tagsMarkdownPreview([
+				{
+					name: 'example',
+					text: 'Not code\n```\ncode();\n```'
+				}
+			], noopToResource),
+			'*@example*  \nNot code\n```\ncode();\n```'
+		);
+	});
+
+	test('Should render @example blocks as code if they contain a <caption>', () => {
+		assert.strictEqual(
+			tagsMarkdownPreview([
+				{
+					name: 'example',
+					text: '<caption>Not code</caption>\ncode();'
+				}
+			], noopToResource),
+			'*@example*  \nNot code\n```\ncode();\n```'
+		);
+	});
+
+	test('Should not render @example blocks as code if they contain a <caption> and a codeblock', () => {
+		assert.strictEqual(
+			tagsMarkdownPreview([
+				{
+					name: 'example',
+					text: '<caption>Not code</caption>\n```\ncode();\n```'
+				}
+			], noopToResource),
+			'*@example*  \nNot code\n```\ncode();\n```'
+		);
+	});
+
 	test('Should render @linkcode symbol name as code', async () => {
 		assert.strictEqual(
 			plainWithLinks([
-				{ "text": "a ", "kind": "text" },
```

Comment:
These changes don't seem directly related to this PR so please revert them

Reply:
This fixes lint warnings, and was immediately next to the text I was editing...

Reply:
Unrelated changes like this make the PRs more difficult to review and muddies up what the commit fixes. Please put it in a separate PR

Reply:
I mean, we've spent more time talking about it than reviewing it would have taken... but sure, seems like a recipe for the codebase getting worse over time to me, but if that's what it takes on a project this size... I'm not going to do it.

Reply:
No need for comments like that. Scoped PRs are just good open source manner and now you know for the future :)